I have seen video but it was very lengthy and complex. While I was doing calmcode course, i have seen using Linear regression and KNN approaches to construct models in Scikit learn course. I googled and found that we can construct "Multi-class classification model" using KNN approach. First, I uploaded the csv file to google colab manually using file sub-option on the left.

Steps we have done:
1. We decided on labels. We took "INDIAN, FRENCH,
 JAPANESE, ITALIAN, MEXICAN, CHINESE, AMERICAN' labels.
2. Then we have annotated labels using prodigy. We used the same textcat.manual but to prevent the multi labels, we used this command "python -m prodigy textcat.manual cuisine_dataset C:\Users\abikki\Documents\HomeWork2\data\homework2_train.jsonl --label INDIAN,FRENCH,JAPANESE,ITALIAN,MEXICAN,CHINESE,AMERICAN --exclusive
3. We have used 'exclusive' to pick only one label as there was both "RELEVANT" and "IRRELEVANT" with multi lables for the dataset. We used the training dataset and took some entries, labelled them and exported them as .json dataset.
4. While annotating, immediately after selecting cuisine label, the value was accepted and we were not able to accept the values without giving labels,so we rejected unrrelevant data and dishes of other cuisines.
5. We then converted the json dataset into csv dataset using an online tool.
6. When we opened the csv file in MS Excel it was having uncessesary data like rejected entries with no cuisine names, session ids and labels. We even removed rejected entries.
7. We removed all those data and kept only two columns.One is text and other is cuisine name.
8. Then we contructed the basic model using KNN Approach after browsing internet to decide betwwen linear regressor or KNN.
9. Doing calmcode course (Scikit learn) gave me idea to try constructing model using the Linear regressor and KNN instead of doing it in more complex and harder way as the YouTube video sited in the proposal document.
10. Due to lack of time, we did not took lot of examples and due to another assignment, we were not able to submit this code before 12.00 but we tried and attaching this.
11. We just wanted to code and build multi-classification model so that we get bonus points. We are happy and striving to learn and improve.

In [4]:
import pandas as pd

# importing csv file as a dataframe for flexibility
df = pd.read_csv('cuisine_dataset.csv')

In [5]:
# displaying the first few rows of the DataFrame
print(df.head())


                                                text   cuisine
0  Your sauce recipe has a lot of starch for the ...   CHINESE
1  I like to dip saltines into Lipton tea that ha...   CHINESE
2  Do they? I know very little about cooking curr...    INDIAN
3  This is really good: Smoky Lentil Chili https:...  AMERICAN
4  That's basically what I did :) I had been eati...  AMERICAN


In [7]:
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer

# 'text' is our text data (comments) and 'cuisine' is our target variable (cuisine names)
vectorizer = TfidfVectorizer()
X = vectorizer.fit_transform(df['text'])
y = df['cuisine']

# Split the data into training and test sets (training data-80% and testing data-20%)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [8]:
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score

# creating an KNN classifier. Randomly taking n=3
knn = KNeighborsClassifier(n_neighbors=3)

# training the model
knn.fit(X_train, y_train)

# predicting on the test set
y_pred = knn.predict(X_test)

# evaluating the model
print(f"Accuracy: {accuracy_score(y_test, y_pred)}")


Accuracy: 0.35


We got an accuracy of 35%. We did not focused on accuracy and we took just around 100 examples and we focused on constructing the model and tried to develop a multi class model for bonus marks. In that 100 data also, we splitted them into 80 (training data) and 20 (testing data).